File: 2350-find-closest-number-to-zero/find-closest-number-to-zero.py
class Solution:
    def findClosestNumber(self, nums: List[int]) -> int:
        
        closest_num = nums[0]
       
        for num in nums:
            if abs(num) < abs(closest_num):
                closest_num = num 
        
        if closest_num < 0 and abs(closest_num) in nums:
            return abs(closest_num)
        
        return closest_num

File: 2350-find-closest-number-to-zero/find-closest-number-to-zero.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findClosestNumber = function (nums) {
    let closest = nums[0];
    //Looping through O(N)
    for (let i = 0; i < nums.length; i++) {
        if (Math.abs(nums[i]) < Math.abs(closest)) {
            closest = nums[i];
        }
    }
                        //Looping again O(N)
    if (closest < 0 && nums.includes(Math.abs(closest))) {
        return Math.abs(closest);
    }else {
        return closest;
    }

    // TC: O(2N) -> O(N)
    // SC: O(1)


};